dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -347,28 +352,22 @@ public String extractFile(InputStream inputStream, String fileType) {
 
         StringBuilder sb = new StringBuilder();
         sb.append(("<root>"));
-        List<String> allFiles = filePaths;
+        Iterator<String> iterator = filePaths.iterator();
         int fileCounter = 0;
-        for (String filePath : allFiles) {
-            if (!filePath.isEmpty() && filePath.length() > 3) {
+        while ((iterator.hasNext() && fileCounter < maxPreviewCount)) {
+            String filePath = iterator.next();
+
+            if (filePath.length() > 3) {
```

Comment:
explain this plz

Reply:
I do not know exactly, this is part of Lock's logic. Should I dig into it? This issue wasn't about this lines of code.

Reply:
Added comments and refactored